spec/requests/start_controller_spec.rb

```diff
@@ -16,29 +15,350 @@
+
+            expected_headers = [
+              "#{current_year} to #{current_year + 1} Logs",
+              "#{current_year - 1} to #{current_year} Logs (Closed collection year)",
+            ]
+
+            expect(logs_headers).to eq expected_headers
+          end
+        end
+
+        context "when the user's org has never submitted sales" do
+          it "does not display data related to sales" do
+            get root_path
+
+            databox_texts = page.find_all(".app-data-box__upper").map(&:text)
+            any_sales_boxes = databox_texts.map(&:downcase).any? { |text| text.include? "sales" }
+
+            expect(any_sales_boxes).to be false
+          end
+        end
+
+        context "when the user's org has submitted sales logs" do
+          before do
+            create(:sales_log, created_by: user)
+          end
+
+          it "displays data related to sales" do
+            get root_path
+
+            databox_texts = page.find_all(".app-data-box__upper").map(&:text)
+            any_sales_boxes = databox_texts.map(&:downcase).any? { |text| text.include? "sales" }
+
+            expect(any_sales_boxes).to be true
+          end
+        end
+
+        context "when the user is a data provider" do
+          let(:user) { create(:user, :data_provider) }
+
+          it "does not display data related to schemes" do
+            get root_path
+
+            databox_texts = page.find_all(".app-data-box__upper").map(&:text)
+            any_schemes_boxes = databox_texts.map(&:downcase).any? { |text| text.include? "schemes" }
+
+            expect(any_schemes_boxes).to be false
+          end
+        end
+
+        context "when the user is a data coordinator" do
+          let(:user) { create(:user, :data_coordinator) }
+
+          it "does display data related to schemes" do
+            get root_path
+
+            databox_texts = page.find_all(".app-data-box__upper").map(&:text)
+            any_schemes_boxes = databox_texts.map(&:downcase).any? { |text| text.include? "schemes" }
+
+            expect(any_schemes_boxes).to be true
+          end
+        end
+
+        context "when the user is a support user" do
+          let(:user) { create(:user, :support) }
+
+          it "does display data related to schemes" do
+            get root_path
+
+            databox_texts = page.find_all(".app-data-box__upper").map(&:text)
+            any_schemes_boxes = databox_texts.map(&:downcase).any? { |text| text.include? "schemes" }
+
+            expect(any_schemes_boxes).to be true
+          end
+        end
+
+        context "when testing the links in the data boxes" do
+          let(:user) { create(:user, :data_coordinator) }
+          let(:in_crossover_period) { true }
+
+          before do
+            create(:sales_log, created_by: user)
+            get root_path
+          end
+
+          it "shows the correct links for in progress lettings" do
+            type = "lettings"
```

Comment:
Where these are the same tests with just the type and status variables differing (particularly as you've already pulled them out), you could parameterise the tests and loop over the combinations rather than duplicating code (see e.g. financial_validations_spec - look for `it "does allow charges inside the range when period is #{test_case[:period][:label]}"`

Reply:
affirmative